contents/Documentation/Effects.playground/Pages/Suspending side effects.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,189 @@
+// nef:begin:header
+/*
+ layout: docs
+ title: Suspending side effects
+ */
+// nef:end
+// nef:begin:hidden
+import Foundation
+import Bow
+import BowEffects
+// nef:end
+/*:
+ # Suspending side effects
+ 
+ {:.beginner}
+ beginner
+ 
+ In Functional Programming, functions must honor its mathematical definition: they must be total, deterministic and pure. However, this is not the situation in many cases; in fact, programs are useful when side-effects happen. The issue is not having side-effects *per se*, but the moment and the circumstances when they are executed. If they happen in the moment we invoke them, reasoning about their outcome becomes much more difficult. In addition to that, composition becomes harder, if not impossible, to achieve.
+ 
+ ## Motivation
+ 
+ Consider the following two functions:
+ */
+func greet(name: String) {
+    print("Hello \(name)!")
+}
+
+func homePage(callback: @escaping (Either<Error, Data>) -> ()) {
+    if let url = URL(string: "https://bow-swift.io") {
+        URLSession.shared.dataTask(with: url) { data, _, error in
+            if let data = data {
+                callback(.right(data))
+            } else if let error = error {
+                callback(.left(error))
+            }
+        }.resume()
+    }
+}
+/*:
+ Function `greet` has a side effect (printing to the console) and does not produce any value, making it harder to compose this function with any other function. Function `homePage` does produce values, but they are returned through a callback as it is an asynchronous operation, making it difficult to be composed as well, besides having another side effect (network call).
+ 
+ ## Creating an IO from a synchronous call
+ 
+ Dealing with this type of functions is usual in our programming routine, either because we have some non-functional legacy code or because we are using libraries that do not provide a functional API. What can we do to overcome this issue?
+ 
+ Bow Effects provides the IO data type. It is a data type that lets us suspend the execution of side effects, providing us a value that describes the side effect, but it has not been performed yet. That way, we can convert our programs into values that we can combine and compose.
+ 
+ `IO` has a method called `invoke` that can help us suspend a side effect. For instance, the `greet` function above can be rewritten as:
+ */
+func greetIO(name: String) -> IO<Never, Void> {
```

Comment:
what do you think to change it to `Task<Void` in order to show another IO options

Reply:
I thought about it but it can be a bit confusing for newcomers. I wanted to only use `IO` in the entire documentation and let the aliases as enhancements that more experienced users can use.